src/formulas/tokenizer.ts

```diff
@@ -223,10 +213,50 @@ function tokenizeSpace(chars: string[]): Token | null {
   return null;
 }
 
-function tokenizeInvalidRange(chars: string[]): Token | null {
-  if (startsWith(chars, INCORRECT_RANGE_STRING)) {
-    chars.splice(0, INCORRECT_RANGE_STRING.length);
+function tokenizeInvalidRange(chars: TokenizingChars): Token | null {
+  if (chars.currentStartsWith(INCORRECT_RANGE_STRING)) {
+    chars.slice(INCORRECT_RANGE_STRING.length);
     return { type: "INVALID_REFERENCE", value: INCORRECT_RANGE_STRING };
   }
   return null;
 }
+
+class TokenizingChars {
+  private chars: string[];
+  private text: string;
+  private currentIndex: number = 0;
+
+  constructor(text: string) {
+    this.text = text;
+    this.chars = text.split("");
+  }
+
+  current() {
+    return this.chars[this.currentIndex];
+  }
+
+  shift() {
+    return this.chars[this.currentIndex++];
+  }
+
+  slice(length: number) {
```

Comment:
I never know what slice does.
Slice, splice... roll the dice :game_die: 

What do you think of renaming this to something like `advanceBy` or `consume` or `forward` or `moveForward` ?

Reply:
ok for advanceBy